### PR TITLE
chore: [IA-249] Add an alert explaining why a permission is required on Android

### PR DESCRIPTION
**Pull request by fabriziofff**
## Short description
This pr adds an alert explaining why a permission is required, before asking for that permission.

https://user-images.githubusercontent.com/26501317/133644033-fdf1fe50-0c1d-4275-b96c-fff8d8471875.mov

Gallery:

Rationale            |  Permission
:-------------------------:|:-------------------------:
 |  

Camera:

Rationale            |  Permission
:-------------------------:|:-------------------------:
 |  

## List of changes proposed in this pull request
- Added `AsyncAlert` that allows to work with `Alert.alert` without callbacks.
- Changed `hasAndroidPermission` in `requestIOAndroidPermission`; now before asking for permission display an alert with the rationale
- Changed `ts/screens/wallet/payment/ScanQrCodeScreen.tsx` in order to display an alert with the rationale before the package `react-native-qrcode-scanner` asks for permission.

## How to test
- Revoke all the permission from the app / re-install
- Messages > EU Covid cert > Save the pass > the rationale alert should be displayed
- Wallet > Pay notice > the rationale alert should be displayed

### CODE REVIEW

File: ios/ItaliaApp/it.lproj/InfoPlist.strings [hunk 1]
```diff
@@ -1,6 +1,6 @@
-"NSCameraUsageDescription" = "IO ha necessità di accedere alla fotocamera per la scansione dei codici di pagamento";
+"NSCameraUsageDescription" = "Potrai leggere codici QR e pagare gli avvisi pagoPA.";
```
Comment (Undermaken) [inline ios/ItaliaApp/it.lproj/InfoPlist.strings:1]: I set my language device to english and I get a message that is not sync with the italian one

Perhaps should we update these too?
https://github.com/pagopa/io-app/blob/9fbe14ea84b2e3246c92b32a3eeff899c8a0622c/ios/ItaliaApp/Info.plist#L62

File: ts/utils/share.ts [hunk 2]
```diff
@@ -42,8 +43,13 @@ export const isShareEnabled = () =>
...
   const hasPermission = tryCatch(
     () =>
-      hasAndroidPermission(
-        PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE
+      requestIOAndroidPermission(
+        PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+        {
+          title: I18n.t("permissionRationale.storage.title"),
+          message: I18n.t("permissionRationale.storage.message"),
+          buttonPositive: I18n.t("global.buttons.choose")
```
Comment (Undermaken) [inline ts/utils/share.ts:51]: It is just a check: I expect `ok` in place of `scegli`
Is it what we want?

File: ts/screens/wallet/payment/ScanQrCodeScreen.tsx [hunk 3]
```diff
@@ -220,6 +232,28 @@ class ScanQrCodeScreen extends React.Component<Props, State> {
     }
   }
 
+  public async componentDidMount() {
+    if (Platform.OS !== "android") {
+      return;
+    }
+    const hasPermission = await PermissionsAndroid.check(
```
Comment (Undermaken) [inline ts/screens/wallet/payment/ScanQrCodeScreen.tsx:239]: It is just a warning/reminder: we display the rational also when the user chose "never ask again"
But we can't know his/her preference without asking the permission

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ios/ItaliaApp/it.lproj/InfoPlist.strings (hunk 1 above)
Comment (fabriziofff) [reply ios/ItaliaApp/it.lproj/InfoPlist.strings:1]: Yes, changed in 7c67d70

File: ts/utils/share.ts (hunk 2 above)
Comment (fabriziofff) [reply ts/utils/share.ts:51]: Yes this is intended by design.

Comment (thisisjp) [reply ts/utils/share.ts:51]: legal design aka @CaraDune :D

**Review by Undermaken — APPROVED** [review]

**Pull request merged**